pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -333,21 +346,42 @@ func (c *Controller) UpdateWorkloadEntryHealth(proxy *model.Proxy, event HealthE
 		return
 	}
 
-	// get previous status
-	cfg := c.store.Get(gvk.WorkloadEntry, entryName, proxy.Metadata.Namespace)
-	if cfg == nil {
-		log.Errorf("config was nil when getting WorkloadEntry %v for %v", entryName, proxy.ID)
-		return
-	}
+	condition := transformHealthEvent(event)
+	c.storeHealthStatus(makeProxyKey(proxy), condition)
```

Comment:
Why do we need to store it in the controller globally and then fetch it later in the same function? Couldn't we just use `condition` directly on L368?

Reply:
I guess maybe if a new event comes in, then we will start off a new goroutine, so we need to use the latest condition. However, I feel like we could approach this different so we don't spin off multiple goroutines all trying to write to the same WE and instead consolidate them?

Reply:
looking